Sources/AppFolder/BaseFolder.swift

```diff
@@ -8,55 +8,95 @@
 
 import Foundation
 
+/// Protocol that define default methods of a Folder.
 public protocol BaseFolder {
     
+    /// Returns the base URL.
     static var baseURL: URL { get }
-    
 }
 
 fileprivate typealias Lib = Library
 fileprivate typealias Docs = Documents
 
 extension BaseFolder {
     
+    /**
+     A reference to the library folder used by this file system.
+     
+     Example:
+     ```
+     let applicationSupport = AppFolder.Library.Application_Support
+     let caches = AppFolder.Library.Caches
+     ```
+     */
     public static var Library: Library {
         return Lib(baseURL: baseURL)
     }
-    
+
+    /**
+     A reference to the document folder used by this file system.
+     
+     # Important # 
+      - Only documents and other data that is user-generated, or that cannot otherwise be recreated by your application, should be stored here and will be automatically backed up by iCloud. 
+     */
     public static var Documents: Documents {
         return Docs(baseURL: baseURL)
     }
     
     #if os(iOS) || os(tvOS) || os(watchOS)
+    /**
+     A reference to the temporary folder used by this file system
+     
+     # Important #
+     - Use this directory to write temporary files that do not need to persist between launches of your app. Your app should remove files from this directory when they are no longer needed however, the system may purge this directory when your app is not running.
+     */
     public static var tmp: Temporary {
-    return Temporary(baseURL: baseURL)
+        return Temporary(baseURL: baseURL)
     }
     #elseif os(macOS)
     @available(*, deprecated, message: "AppFolder.tmp is unavailable on macOS")
+    /**
+     A reference to the temporary folder used by this file system
+     
+     # Important #
+      - Use this directory to write temporary files that do not need to persist between launches of your app. Your app should remove files from this directory when they are no longer needed however, the system may purge this directory when your app is not running.
+     */
     public static var tmp: Temporary {
         return Temporary(baseURL: baseURL)
     }
     #endif
     
 }
 
-public enum AppFolder : BaseFolder {
+/**
+ 
+ A Representation of a directories inside your app's container
+ 
+ ````
+ let documents = AppFolder.Documents
+ let caches = AppFolder.Library.Caches
+ let cachesURL = caches.url
+ ````
+ 
+ */ 
+public enum AppFolder: BaseFolder {
     
+    /// Returns the base URL.
     public static let baseURL: URL = {
         let url = URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true)
         return url
     }()
     
 }
 
+/// Protocol that define default methods of a AppGroup.
 public protocol AppGroup {
-    
     static var groupIdentifier: String { get }
-    
```

Comment:
I’m also talking about this: it’s small and I don’t have very strong feelings about this, but still its unrelated to the scope of this pull request